### PR TITLE
Set default volume and concentration units

### PR DESCRIPTION
**Pull request by djcooke**

### CODE REVIEW

**Comment by djcooke** [comment]
* set default units
* updated all unit titles to be plural
* set units to null wherever corresponding field was null
* set units to default wherever the field was filled, but no units were saved
* added validation to ensure units are always provided when the corresponding field is filled
* fixed poolable elements list to show unit abbreviation instead of enum item name

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/ConcentrationUnit.java [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 public enum ConcentrationUnit {
 
-  NANOGRAMS_PER_MICROLITRE("ng/&#181;L"), NANOMOLAR("nM");
+  NANOGRAMS_PER_MICROLITRE("ng/&#181;L"), NANOMOLARS("nM");
```
Comment (callunity) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/ConcentrationUnit.java]: Should remain `NANOMOLAR`

File: sqlstore/src/main/resources/db/migration/V8503__volume_concentration_units.sql [hunk 2]
```diff
@@ -0,0 +1,27 @@
+UPDATE Sample SET volumeUnits = NULL WHERE volume IS NULL;
+UPDATE Sample SET concentrationUnits = NULL WHERE concentration IS NULL;
+UPDATE Sample SET volumeUnits = 'MICROLITRES' WHERE volumeUnits = 'MICROLITRE';
+UPDATE Sample SET volumeUnits = 'MICROLITRES' WHERE volume IS NOT NULL AND volumeUnits IS NULL;
+UPDATE Sample SET concentrationUnits = 'NANOMOLARS' WHERE concentrationUnits = 'NANOMOLAR';
```
Comment (callunity) [inline sqlstore/src/main/resources/db/migration/V8503__volume_concentration_units.sql]: Should remain NANOMOLAR

Comment (djcooke) [reply sqlstore/src/main/resources/db/migration/V8503__volume_concentration_units.sql]: Fixed

**Comment by sqwci** [comment]
SonarQube analysis reported 23 issues
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 11 major
* ![MINOR][MINOR] 10 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [DefaultSampleService.java#L236](https://github.com/TGAC/miso-lims/blob/7010140cea5266d8f69ae24dfdf6494a3c0aa9f0/miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java#L236): Refactor this method to reduce its Cognitive Complexity from 30 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [DefaultSampleService.java#L299](https://github.com/TGAC/miso-lims/blob/7010140cea5266d8f69ae24dfdf6494a3c0aa9f0/miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java#L299): Refactor this method to reduce its Cognitive Complexity from 18 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [DefaultSampleService.java#L94](https://github.com/TGAC/miso-lims/blob/7010140cea5266d8f69ae24dfdf6494a3c0aa9f0/miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java#L94): A "NullPointerException" could be thrown; "parent" is nullable here. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)
1. ![MAJOR][MAJOR] [DefaultSampleService.java#L417](https://github.com/TGAC/miso-lims/blob/7010140cea5266d8f69ae24dfdf6494a3c0aa9f0/miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java#L417): Format specifiers should be used instead of string concatenation. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3457)
1. ![MAJOR][MAJOR] [hot_library.js#L68](https://github.com/TGAC/miso-lims/blob/7010140cea5266d8f69ae24dfdf6494a3c0aa9f0/miso-web/src/main/webapp/scripts/hot_library.js#L68): Rename "pt" as this name is already used in declaration at line 47. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2814)
1. ![MAJOR][MAJOR] [hot_library.js#L72](https://github.com/TGAC/miso-lims/blob/7010140cea5266d8f69ae24dfdf6494a3c0aa9f0/miso-web/src/main/webapp/scripts/hot_library.js#L72): Rename "indices" as this name is already used in declaration at line 51. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2814)
1. ![MAJOR][MAJOR] [hot_library.js#L82](https://github.com/TGAC/miso-lims/blob/7010140cea5266d8f69ae24dfdf6494a3c0aa9f0/miso-web/src/main/webapp/scripts/hot_library.js#L82): Rename "pt" as this name is already used in declaration at line 47. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2814)
1. ![MAJOR][MAJOR] [hot_sample.js#L341](https://github.com/TGAC/miso-lims/blob/7010140cea5266d8f69ae24dfdf6494a3c0aa9f0/miso-web/src/main/webapp/scripts/hot_sample.js#L341): Rename "renderer" as this name is already used in declaration at line 337. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2814)
1. ![MINOR][MINOR] [DefaultSampleService.java#L413](https://github.com/TGAC/miso-lims/blob/7010140cea5266d8f69ae24dfdf6494a3c0aa9f0/miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java#L413): Use isEmpty() to check whether the collection is empty or not. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1155)
1. ![MINOR][MINOR] [DefaultSampleService.java#L473](https://github.com/TGAC/miso-lims/blob/7010140cea5266d8f69ae24dfdf6494a3c0aa9f0/miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java#L473): Remove the declaration of thrown exception 'org.hibernate.exception.ConstraintViolationException' which is a runtime exception. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by sqwci** [comment]
SonarQube analysis reported 23 issues
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 11 major
* ![MINOR][MINOR] 10 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [DefaultSampleService.java#L236](https://github.com/TGAC/miso-lims/blob/9ad56274cff45d1448d65671e867de7a9b05546e/miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java#L236): Refactor this method to reduce its Cognitive Complexity from 30 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [DefaultSampleService.java#L299](https://github.com/TGAC/miso-lims/blob/9ad56274cff45d1448d65671e867de7a9b05546e/miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java#L299): Refactor this method to reduce its Cognitive Complexity from 18 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [DefaultSampleService.java#L94](https://github.com/TGAC/miso-lims/blob/9ad56274cff45d1448d65671e867de7a9b05546e/miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java#L94): A "NullPointerException" could be thrown; "parent" is nullable here. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)
1. ![MAJOR][MAJOR] [DefaultSampleService.java#L417](https://github.com/TGAC/miso-lims/blob/9ad56274cff45d1448d65671e867de7a9b05546e/miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java#L417): Format specifiers should be used instead of string concatenation. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3457)
1. ![MAJOR][MAJOR] [hot_library.js#L68](https://github.com/TGAC/miso-lims/blob/9ad56274cff45d1448d65671e867de7a9b05546e/miso-web/src/main/webapp/scripts/hot_library.js#L68): Rename "pt" as this name is already used in declaration at line 47. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2814)
1. ![MAJOR][MAJOR] [hot_library.js#L72](https://github.com/TGAC/miso-lims/blob/9ad56274cff45d1448d65671e867de7a9b05546e/miso-web/src/main/webapp/scripts/hot_library.js#L72): Rename "indices" as this name is already used in declaration at line 51. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2814)
1. ![MAJOR][MAJOR] [hot_library.js#L82](https://github.com/TGAC/miso-lims/blob/9ad56274cff45d1448d65671e867de7a9b05546e/miso-web/src/main/webapp/scripts/hot_library.js#L82): Rename "pt" as this name is already used in declaration at line 47. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2814)
1. ![MAJOR][MAJOR] [hot_sample.js#L341](https://github.com/TGAC/miso-lims/blob/9ad56274cff45d1448d65671e867de7a9b05546e/miso-web/src/main/webapp/scripts/hot_sample.js#L341): Rename "renderer" as this name is already used in declaration at line 337. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2814)
1. ![MINOR][MINOR] [DefaultSampleService.java#L413](https://github.com/TGAC/miso-lims/blob/9ad56274cff45d1448d65671e867de7a9b05546e/miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java#L413): Use isEmpty() to check whether the collection is empty or not. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1155)
1. ![MINOR][MINOR] [DefaultSampleService.java#L473](https://github.com/TGAC/miso-lims/blob/9ad56274cff45d1448d65671e867de7a9b05546e/miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java#L473): Remove the declaration of thrown exception 'org.hibernate.exception.ConstraintViolationException' which is a runtime exception. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by sqwci** [comment]
SonarQube analysis reported 23 issues
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 11 major
* ![MINOR][MINOR] 10 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [DefaultSampleService.java#L236](https://github.com/TGAC/miso-lims/blob/d52c2765b92cfbe28e5885b3f348104484216d73/miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java#L236): Refactor this method to reduce its Cognitive Complexity from 30 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [DefaultSampleService.java#L299](https://github.com/TGAC/miso-lims/blob/d52c2765b92cfbe28e5885b3f348104484216d73/miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java#L299): Refactor this method to reduce its Cognitive Complexity from 18 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [DefaultSampleService.java#L94](https://github.com/TGAC/miso-lims/blob/d52c2765b92cfbe28e5885b3f348104484216d73/miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java#L94): A "NullPointerException" could be thrown; "parent" is nullable here. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)
1. ![MAJOR][MAJOR] [DefaultSampleService.java#L417](https://github.com/TGAC/miso-lims/blob/d52c2765b92cfbe28e5885b3f348104484216d73/miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java#L417): Format specifiers should be used instead of string concatenation. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3457)
1. ![MAJOR][MAJOR] [hot_library.js#L68](https://github.com/TGAC/miso-lims/blob/d52c2765b92cfbe28e5885b3f348104484216d73/miso-web/src/main/webapp/scripts/hot_library.js#L68): Rename "pt" as this name is already used in declaration at line 47. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2814)
1. ![MAJOR][MAJOR] [hot_library.js#L72](https://github.com/TGAC/miso-lims/blob/d52c2765b92cfbe28e5885b3f348104484216d73/miso-web/src/main/webapp/scripts/hot_library.js#L72): Rename "indices" as this name is already used in declaration at line 51. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2814)
1. ![MAJOR][MAJOR] [hot_library.js#L82](https://github.com/TGAC/miso-lims/blob/d52c2765b92cfbe28e5885b3f348104484216d73/miso-web/src/main/webapp/scripts/hot_library.js#L82): Rename "pt" as this name is already used in declaration at line 47. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2814)
1. ![MAJOR][MAJOR] [hot_sample.js#L341](https://github.com/TGAC/miso-lims/blob/d52c2765b92cfbe28e5885b3f348104484216d73/miso-web/src/main/webapp/scripts/hot_sample.js#L341): Rename "renderer" as this name is already used in declaration at line 337. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2814)
1. ![MINOR][MINOR] [DefaultSampleService.java#L413](https://github.com/TGAC/miso-lims/blob/d52c2765b92cfbe28e5885b3f348104484216d73/miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java#L413): Use isEmpty() to check whether the collection is empty or not. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1155)
1. ![MINOR][MINOR] [DefaultSampleService.java#L473](https://github.com/TGAC/miso-lims/blob/d52c2765b92cfbe28e5885b3f348104484216d73/miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java#L473): Remove the declaration of thrown exception 'org.hibernate.exception.ConstraintViolationException' which is a runtime exception. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by sqwci** [comment]
SonarQube analysis reported 23 issues
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 11 major
* ![MINOR][MINOR] 10 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [DefaultSampleService.java#L236](https://github.com/TGAC/miso-lims/blob/62ea7bd5302a310cd0123b89c2d5fa87e67acf18/miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java#L236): Refactor this method to reduce its Cognitive Complexity from 30 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [DefaultSampleService.java#L299](https://github.com/TGAC/miso-lims/blob/62ea7bd5302a310cd0123b89c2d5fa87e67acf18/miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java#L299): Refactor this method to reduce its Cognitive Complexity from 18 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [DefaultSampleService.java#L94](https://github.com/TGAC/miso-lims/blob/62ea7bd5302a310cd0123b89c2d5fa87e67acf18/miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java#L94): A "NullPointerException" could be thrown; "parent" is nullable here. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)
1. ![MAJOR][MAJOR] [DefaultSampleService.java#L417](https://github.com/TGAC/miso-lims/blob/62ea7bd5302a310cd0123b89c2d5fa87e67acf18/miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java#L417): Format specifiers should be used instead of string concatenation. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3457)
1. ![MAJOR][MAJOR] [hot_library.js#L68](https://github.com/TGAC/miso-lims/blob/62ea7bd5302a310cd0123b89c2d5fa87e67acf18/miso-web/src/main/webapp/scripts/hot_library.js#L68): Rename "pt" as this name is already used in declaration at line 47. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2814)
1. ![MAJOR][MAJOR] [hot_library.js#L72](https://github.com/TGAC/miso-lims/blob/62ea7bd5302a310cd0123b89c2d5fa87e67acf18/miso-web/src/main/webapp/scripts/hot_library.js#L72): Rename "indices" as this name is already used in declaration at line 51. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2814)
1. ![MAJOR][MAJOR] [hot_library.js#L82](https://github.com/TGAC/miso-lims/blob/62ea7bd5302a310cd0123b89c2d5fa87e67acf18/miso-web/src/main/webapp/scripts/hot_library.js#L82): Rename "pt" as this name is already used in declaration at line 47. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2814)
1. ![MAJOR][MAJOR] [hot_sample.js#L341](https://github.com/TGAC/miso-lims/blob/62ea7bd5302a310cd0123b89c2d5fa87e67acf18/miso-web/src/main/webapp/scripts/hot_sample.js#L341): Rename "renderer" as this name is already used in declaration at line 337. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2814)
1. ![MINOR][MINOR] [DefaultSampleService.java#L413](https://github.com/TGAC/miso-lims/blob/62ea7bd5302a310cd0123b89c2d5fa87e67acf18/miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java#L413): Use isEmpty() to check whether the collection is empty or not. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1155)
1. ![MINOR][MINOR] [DefaultSampleService.java#L473](https://github.com/TGAC/miso-lims/blob/62ea7bd5302a310cd0123b89c2d5fa87e67acf18/miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java#L473): Remove the declaration of thrown exception 'org.hibernate.exception.ConstraintViolationException' which is a runtime exception. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**